pkg/apis/cluster/v1alpha1/types.go

```diff
@@ -116,6 +137,66 @@ type ClusterSpec struct {
 	// any resource that does not tolerate the Taint.
 	// +optional
 	Taints []corev1.Taint `json:"taints,omitempty"`
+
+	// ResourceModels is the list of resource modeling in this cluster. Each modeling quota can be customized by the user.
+	// Modeling name must be one of the following: cpu, memory, storage, ephemeral-storage.
+	// If the user does not define the modeling name and modeling quota, it will be the default model.
+	// The default model index from 0 to 11. Each default model's cpu quota is (2^index-1, 2^index], index >= 0
+	// Each default model's memory quota is also (2^index-1, 2^index], index >= 0, for example, model0 is like this:
+	// - index: 0
+	// ranges:
+	//   - name: CPU
+	//     min: 0 C
+	//     max: 1 C
+	//   - name: memory
+	//     min: 0 GB
+	//     max: 1 GB
+	// model10 is like this:
+	// - index: 10
+	// range:
+	//   - name: CPU
+	//     min: 512 C
+	//     max: 1024 C
+	//   - name: memory
+	//     min: 512 GB
+	//     max: 1024 GB
+	// model11 is like this:
+	// - index: 11
+	// range:
+	//   - name: CPU
+	//     min: 1024 C
+	//     max: MAXINT
+	//   - name: memory
+	//     min: 1024 GB
+	//     max: MAXINT
+	// +optional
+	ResourceModels []ResourceModel `json:"resourceModels,omitempty"`
+}
+
+// ResourceModel describes the modeling that you want to statistics.
+type ResourceModel struct {
+	// Grade is the index for the resource modeling.
+	// +optional
+	Grade int `json:"grade,omitempty"`
+
+	// Ranges describes the resource quota ranges.
+	// +optional
+	Ranges []ResourceModelItem `json:"ranges,omitempty"`
+}
+
+// ResourceModelItem describes the detail of each modeling quota that ranges from min to max.
+type ResourceModelItem struct {
+	// Name is the name for the resource that you want to categorize.
+	// +optional
+	Name ResourceName `json:"name,omitempty"`
+
+	// Min is the minimum amount of this resource represented by resource name。
+	// +optional
+	Min resource.Quantity `json:"min,omitempty"`
+
+	// Max is the maximum amount of this resource represented by resource name。
+	// +optional
+	Max resource.Quantity `json:"max,omitempty"`
```

Comment:
Do these 3 fields really `optional`? 
What if the user does not set them(or one of them)? 

In addition, how to set the last model range? For example, the user wants to set 2 models,
The first one is used to target the small nodes, like(1c, 10c], and the second one is used to target the larger nodes which have more than `10c`, but don't know the exact `Max` value.

Reply:
The intention of set optional in there is that if the user does not want to set any value, the system will select the default modeling.

For the question2, we can make on a rule: the minimum value here can be obtained, and the maximum value cannot be obtained. In other words, the interval is closed on the left and open on the right. So in your example, the first one will be [1c,10c), the second one will be [10c, ∞).

For the right side of the last interval, which defaults to no matter what value you pass, anything is infinite to me. Because the quota of resources is larger than the left value of the last interval will be classified into the last interval. I have added some comments on this part.